### ISSUE TITLE
Implement the rights to object and restrict

### ISSUE THREAD

**Issue by sindelio** [ClearGDPR/ClearGDPR#100]
We have tasks for:

- [ ] Right to portability
- [ ] Right to rectify

But we can't forget:

- [ ] Right to object
- [ ] Right to restrict

Works on those last have started!

**Comment by sindelio** [comment]
References #20

**Comment by sindelio** [comment]
Right to restrict:
1. - [] Research more about what it means to restrict the processing of data
2. - [] Update the smart contract 
3. - [] Update the blockchain related back-end
4. - [] Create API endpoints in the subjects domain to object the processing of data. This does not require a schema, but will require an update in the possible data status
5. - [] Create UI for the restriction of processing. Need more discussion about this

Right to object:
1. - [] Research more about what it means to object the processing of data
2. - [] Update the smart contract
3. - [] Update the blockchain related back-end
4. - [] Create API endpoints in the subjects domain to object the processing of data
5. - [] Create UI for the objection of data processing. Need more discussion about this

Depends on #140

**Comment by sindelio** [comment]
Art. 16 GDPR - Right to rectification

"The data subject shall have the right to obtain from the controller without undue delay the rectification of inaccurate personal data concerning him or her. Taking into account the purposes of the processing, the data subject shall have the right to have incomplete personal data completed, including by means of providing a supplementary statement."

**Comment by sindelio** [comment]
Art. 18 GDPR - Right to restriction of processing

"1. The data subject shall have the right to obtain from the controller restriction of processing where one of the following applies:
    (a) the accuracy of the personal data is contested by the data subject, for a period enabling the controller to verify the accuracy of the personal data;
    (b) the processing is unlawful and the data subject opposes the erasure of the personal data and requests the restriction of their use instead;
    (c) the controller no longer needs the personal data for the purposes of the processing, but they are required by the data subject for the establishment, exercise or defence of legal claims;
    (d) the data subject has objected to processing pursuant to Article 21(1) pending the verification whether the legitimate grounds of the controller override those of the data subject.
2. Where processing has been restricted under paragraph 1, such personal data shall, with the exception of storage, only be processed with the data subject’s consent or for the establishment, exercise or defence of legal claims or for the protection of the rights of another natural or legal person or for reasons of important public interest of the Union or of a Member State.
3. A data subject who has obtained restriction of processing pursuant to paragraph 1 shall be informed by the controller before the restriction of processing is lifted.

**Comment by sindelio** [comment]
Art. 20 GDPR - Right to data portability

"1. The data subject shall have the right to receive the personal data concerning him or her, which he or she has provided to a controller, in a structured, commonly used and machine-readable format and have the right to transmit those data to another controller without hindrance from the controller to which the personal data have been provided, where:
    (a) the processing is based on consent pursuant to point (a) of Article 6(1) or point (a) of Article 9(2) or on a contract pursuant to point (b) of Article 6(1); and
    (b) the processing is carried out by automated means.
2. In exercising his or her right to data portability pursuant to paragraph 1, the data subject shall have the right to have the personal data transmitted directly from one controller to another, where technically feasible.
3. The exercise of the right referred to in paragraph 1 of this Article shall be without prejudice to Article 17. That right shall not apply to processing necessary for the performance of a task carried out in the public interest or in the exercise of official authority vested in the controller.
4. The right referred to in paragraph 1 shall not adversely affect the rights and freedoms of others."

I think we've already covered everything for this right.

**Comment by sindelio** [comment]
Art. 21 GDPR - Right to object

"1. The data subject shall have the right to object, on grounds relating to his or her particular situation, at any time to processing of personal data concerning him or her which is based on point (e) or (f) of Article 6(1), including profiling based on those provisions. The controller shall no longer process the personal data unless the controller demonstrates compelling legitimate grounds for the processing which override the interests, rights and freedoms of the data subject or for the establishment, exercise or defence of legal claims.
2. Where personal data are processed for direct marketing purposes, the data subject shall have the right to object at any time to processing of personal data concerning him or her for such marketing, which includes profiling to the extent that it is related to such direct marketing.
3. Where the data subject objects to processing for direct marketing purposes, the personal data shall no longer be processed for such purposes.
4. At the latest at the time of the first communication with the data subject, the right referred to in paragraphs 1 and 2 shall be explicitly brought to the attention of the data subject and shall be presented clearly and separately from any other information.
5. In the context of the use of information society services, and notwithstanding Directive 2002/58/EC, the data subject may exercise his or her right to object by automated means using technical specifications.
6. Where personal data are processed for scientific or historical research purposes or statistical purposes pursuant to Article 89(1), the data subject, on grounds relating to his or her particular situation, shall have the right to object to processing of personal data concerning him or her, unless the processing is necessary for the performance of a task carried out for reasons of public interest."

This one is similar to the right to restrict, but it's all or nothing. To object means to stop completely, all processors and the controller will stop processing the data. Whereas for restriction, a subject can restrict a specific processor, for example.

**Comment by sindelio** [comment]
Let's discuss this guys!

**Comment by sindelio** [comment]
Handled by #147  and #148
